sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -141,16 +146,12 @@ def _run_command(command):
             "stderr": subprocess.PIPE,
             "cwd": working_directory,
             "universal_newlines": True,
+            "timeout": 10,
             "env": dict(os.environ, AZURE_CORE_NO_COLOR="true"),
         }
-        if platform.python_version() >= "3.3":
-            kwargs["timeout"] = 10
-
         return subprocess.check_output(args, **kwargs)
     except subprocess.CalledProcessError as ex:
         # non-zero return from shell
-        if ex.returncode == 127 or ex.stderr.startswith("'az' is not recognized"):
```

Comment:
Given 127 is literally the code for "command not found", where we not complicating our life here with an additional string testing, and the fix could be simply test for 127 and nothing else?

Reply:
An error code of 127 is only returned on unix-based systems where `/bin/sh` is run. In windows, where `cmd.exe` is used, the return code from the subprocess is 1 when az is not found. So I am assuming that is why the string was checked initially.

In my investigation I found that adding " & exit" to the end of the command for windows will propagate the correct error code (9009). So an alternative approach could be to do that and then check if `ex.returncode in (127, 9009)`. However, I thought a call to `shutil.which` at the beginning of the function would be more straightforward/efficient.